lib/logger_json/formatter.ex

```diff
@@ -16,6 +23,7 @@ defmodule LoggerJSON.Formatter do
               msg :: Logger.message(),
               ts :: Logger.Formatter.time(),
               md :: [atom] | :all,
-              state :: map
+              state :: map,
+              formatter_state :: Keyword.t()
```

Comment:
I think we should use a map for the state. Keywords have higher complexity for reading operations because sometimes we need to traverse the full list to get a value, which would be harmful as the state grows. Maps are highly optimized for that in Erlang:
```suggestion
              formatter_state :: map()
```